### PR TITLE
Send CloudTrail logs to Sumo Logic

### PR DESCRIPTION
**Pull request by samcrang**

### CODE REVIEW

**Comment by samcrang** [comment]
I found a bug in this so it's probably not a sensible idea to merge this (yet).

File: aws/sumologic/lambda/cloudfront-s3-logs-to-sumologic/test/cloudfront_test.js [hunk 1]
```diff
@@ -19,10 +18,11 @@ describe("Sumo Logic S3 CloudFront Lambda", function() {
 
       AWS.mock("S3", "getObject", new Buffer(TEST_DATA.compressed.cloudfront_logs));
 
-      subject.handler(TEST_DATA.compressed.lambda_event, fake_context(function(err) {
-        assert.equal(err, undefined);
+      subject.handler(TEST_DATA.compressed.lambda_event, {}, function(error, result) {
```
Comment (philandstuff) [inline aws/sumologic/lambda/cloudfront-s3-logs-to-sumologic/test/cloudfront_test.js]: why does this function take two parameters? It looks to me like it only ever gets called with zero or one arguments.

File: aws/sumologic/lambda/cloudfront-s3-logs-to-sumologic/test/cloudfront_test.js [hunk 2]
```diff
@@ -32,10 +32,11 @@ describe("Sumo Logic S3 CloudFront Lambda", function() {
 
       AWS.mock("S3", "getObject", new Buffer(TEST_DATA.uncompressed.cloudfront_logs));
 
-      subject.handler(TEST_DATA.uncompressed.lambda_event, fake_context(function(err) {
-        assert.equal(err, undefined);
+      subject.handler(TEST_DATA.uncompressed.lambda_event, {}, function(error, result) {
```
Comment (philandstuff) [inline aws/sumologic/lambda/cloudfront-s3-logs-to-sumologic/test/cloudfront_test.js]: ditto

**Comment by samcrang** [comment]
This should be good to go now 🐐

File: aws/sumologic/lambda/cloudtrail-s3-logs-to-sumologic/test/cloudtrail_test.js [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+    });
+
+    it("should send compressed logs to Sumo Logic", function(done) {
+      var http_mock = nock('https://endpoint1.collection.eu.sumologic.com')
+        .post("/receiver/v1/http/<XXXX>", TEST_DATA.uncompressed.processed_logs.toString())
+        .reply(200);
+
+      AWS.mock("S3", "getObject", new Buffer(TEST_DATA.compressed.cloudtrail_logs));
+
+      subject.handler(TEST_DATA.compressed.lambda_event, {}, function(error) {
```
Comment (philandstuff) [inline aws/sumologic/lambda/cloudtrail-s3-logs-to-sumologic/test/cloudtrail_test.js]: looks like these tests do it right 😄

File: aws/sumologic/lambda/cloudfront-s3-logs-to-sumologic/cloudfront.js [hunk 4]
```diff
@@ -69,7 +69,7 @@ function s3LogsToSumo(bucket, objKey, context, s3) {
 
     var s3Stream = s3.getObject({ Bucket: bucket, Key: objKey }).createReadStream();
     s3Stream.on('error', function(error) {
-        context.fail(error);
+        callback(error);
```
Comment (philandstuff) [inline aws/sumologic/lambda/cloudfront-s3-logs-to-sumologic/cloudfront.js]: `function(error) { callback(error); }` is bascially the same as `callback`

File: aws/sumologic/lambda/cloudfront-s3-logs-to-sumologic/test/cloudfront_test.js (hunk 1 above)
Comment (samcrang) [inline aws/sumologic/lambda/cloudfront-s3-logs-to-sumologic/test/cloudfront_test.js]: You can use that parameter ([docs](https://docs.aws.amazon.com/lambda/latest/dg/nodejs-prog-model-handler.html#nodejs-prog-model-handler-callback)), but we don't so it's probably sensible to remove it.

File: aws/sumologic/lambda/cloudfront-s3-logs-to-sumologic/cloudfront.js (hunk 4 above)
Comment (samcrang) [inline aws/sumologic/lambda/cloudfront-s3-logs-to-sumologic/cloudfront.js]: 👍

**Pull request merged**